From 2852b8bee2518865512aa4d89cb0e41ed25dee09 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 28 Nov 2024 11:38:01 +0100
Subject: [PATCH] drm/stm: ltdc: fix remove support of yuv pixel formats for
 MP21

Remove planar & semiplanar yuv formats for all layers on MP21.

Change-Id: I23f0d084f10f0aa77db5d556b987e1f335f8fcbf
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/421345
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -1624,11 +1624,11 @@ static struct drm_plane *ltdc_plane_crea
 		}
 
 		/*
-		 * Soc MP21 & MP25 doesn't support pixel formats yuv semiplanar &
-		 * planar on layer1 only.
+		 * Soc MP25 doesn't support YUV semiplanar and planar pixel formats on layer 1.
+		 * Soc MP21 doesn't support YUV semiplanar and planar pixel formats (all layers).
 		 */
-		if (!((of_device_is_compatible(dev->of_node, "st,stm32mp21-ltdc") ||
-		    of_device_is_compatible(dev->of_node, "st,stm32mp25-ltdc")) && !index)) {
+		if ((of_device_is_compatible(dev->of_node, "st,stm32mp25-ltdc") && index) ||
+		    (of_device_is_compatible(dev->of_node, "st,stm32-ltdc"))) {
 			if (val & LXCR_C1R_YSPA) {
 				memcpy(&formats[nb_fmt], ltdc_drm_fmt_ycbcr_sp,
 				       ARRAY_SIZE(ltdc_drm_fmt_ycbcr_sp) * sizeof(*formats));
